src/HDF5.jl

```diff
@@ -1819,8 +1818,8 @@ function h5d_write(dataset_id::hid_t, memtype_id::hid_t, v::VLen{T}, xfer::hid_t
 end
 h5f_create(filename::String) = h5f_create(filename, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT)
 h5f_open(filename::String, mode) = h5f_open(filename, mode, H5P_DEFAULT)
-h5g_create(obj_id::hid_t, name::String) = h5g_create(obj_id, name, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT)
-h5g_create(obj_id::hid_t, name::String, lcpl_id, gcpl_id) = h5g_create(obj_id, name, lcpl_id, gcpl_id, H5P_DEFAULT)
+h5g_create(obj_id, name) = h5g_create(obj_id, name, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT)
+h5g_create(obj_id, name, lcpl_id, gcpl_id) = h5g_create(obj_id, name, lcpl_id, gcpl_id, H5P_DEFAULT)
```

Comment:
should we get rid of these methods and call `h5g_create` in `g_create` with the defaults supplied?

Reply:
I guess that might be annoying for users that directly access this function to now be forced to explicitly need to pass all these arguments. In any case, probably follow up work anyways.

Reply:
I think we should (but it should be it's own PR) — that was part of a plan I've had in my head to better differentiate the middle-level from low-level.